output/streaming/instant.go

```diff
@@ -0,0 +1,150 @@
+package streaming
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/cube2222/octosql"
+	"github.com/cube2222/octosql/execution"
+	"github.com/cube2222/octosql/storage"
+)
+
+var outputRecordsPrefix = []byte("$output_records$")
+var errorPrefix = []byte("$error$")
+var endOfStreamPrefix = []byte("$end_of_stream$")
+
+type InstantStreamOutput struct {
+	StreamID *execution.StreamID
+}
+
+func NewInstantStreamOutput(streamID *execution.StreamID) *InstantStreamOutput {
+	return &InstantStreamOutput{
+		StreamID: streamID,
+	}
+}
+
+func (o *InstantStreamOutput) ReadyForMore(ctx context.Context, tx storage.StateTransaction) error {
+	return nil
+}
+
+func (o *InstantStreamOutput) AddRecord(ctx context.Context, tx storage.StateTransaction, inputIndex int, record *execution.Record) error {
+	tx = tx.WithPrefix(o.StreamID.AsPrefix())
+	if inputIndex != 0 {
+		return errors.Errorf("only one input stream allowed for output, got input index %d", inputIndex)
+	}
+
+	outputRecords := execution.NewOutputQueue(tx.WithPrefix(outputRecordsPrefix))
+	if err := outputRecords.Push(ctx, record); err != nil {
+		return errors.Wrap(err, "couldn't push current record to output queue")
+	}
+
+	return nil
+}
+
+func (o *InstantStreamOutput) Next(ctx context.Context, tx storage.StateTransaction) (*execution.Record, error) {
+	tx = tx.WithPrefix(o.StreamID.AsPrefix())
+	// If no element to get and end of stream then end of stream else wait. Also check error.
+	records := execution.NewOutputQueue(tx.WithPrefix(outputRecordsPrefix))
+	var record execution.Record
+	err := records.Pop(ctx, &record)
```

Comment:
also maybe newline here, this is kinda too much in 1 place for me

Reply:
or 1 line higher to keep record declaration and Pop together (whichever you prefer)